File: examples/swift/notes-showcase/Tests/NotesShowcaseTests/ThemeAdapterTests.swift
import XCTest
import SwiftUI
import VMx
@testable import NotesShowcase
@testable import NotesShowcaseCore

final class ThemeAdapterTests: XCTestCase {
    func testIndependentHighContrastFlagProjectsBlackWhitePalette() throws {
        let initial = ThemeModel(
            name: "light",
            accentColor: "#ABCDEF",
            fontScaleFactor: 1.0,
            highContrast: true,
            followsSystem: false
        )
        let vm = try ThemeVM.builder()
            .name("theme")
            .services(hub: MessageHub(), dispatcher: ImmediateDispatcher.INSTANCE)
            .initialModel(initial)
            .build()
        try vm.construct()

        let adapter = ThemeAdapter(themeVM: vm)

        XCTAssertEqual(adapter.background, Color.black)
        XCTAssertEqual(adapter.pane, Color.black)
        XCTAssertEqual(adapter.textDim, Color.white)
        vm.dispose()
    }
}

File: examples/csharp/avalonia/NotesShowcase.Tests/Views/Adapter/ThemeAdapterTests.cs
using Avalonia;
using Avalonia.Controls;
using Avalonia.Headless.XUnit;
using Avalonia.Media;
using Avalonia.Styling;
using NotesShowcase.Models;
using NotesShowcase.ViewModels;
using NotesShowcase.Views.Adapter;
using System.Reactive.Concurrency;
using VMx.Services;
using Xunit;

namespace NotesShowcase.Tests.Views.Adapter;

public sealed class ThemeAdapterTests
{
    [AvaloniaFact]
    public void Independent_high_contrast_toggle_projects_black_white_palette()
    {
        var vm = ThemeVM.Builder()
            .Name("theme")
            .Services(
                new MessageHub(),
                new RxDispatcher(ImmediateScheduler.Instance, ImmediateScheduler.Instance))
            .InitialModel(ThemeModel.LIGHT_PRESET)
            .Build();
        vm.Construct();
        var app = Application.Current!;
        using var binding = ThemeAdapter.Bind(vm, app);

        vm.ToggleHighContrast.Execute(null);

        var palette = Assert.IsType<ResourceDictionary>(app.Resources.MergedDictionaries[^1]);
        Assert.Equal(Color.Parse("#FF000000"), palette["BgColor"]);
        Assert.Equal(Color.Parse("#FFFFFFFF"), palette["TextColor"]);
        Assert.Equal(Color.Parse("#FFFFFFFF"), palette["BorderColor"]);
        Assert.Equal(ThemeVariant.Dark, app.RequestedThemeVariant);

        vm.Dispose();
    }
}
